### PR TITLE
Encode QoL

### PR DESCRIPTION
**Pull request by uncomputable**
Improve quality of life in the encode module.

### CODE REVIEW

File: src/bit_encoding/bitwriter.rs [hunk 1]
```diff
@@ -109,3 +109,42 @@ impl<W: io::Write> BitWriter<W> {
         Ok(len)
     }
 }
+
+// Implement for io::Sink to make vec() an associated function
+// Implementing for a generic W would require callers to specify W, which is annoying
```
Comment (apoelstra) [inline src/bit_encoding/bitwriter.rs:114]: In fc91f4e32788f03d6d303fdc681704c68a3065a9:

This "implement on `BitWriter<io::Sink>` trick is pretty clever but I think it's too clever :). We should just make this a standalone function.

**Comment by apoelstra** [comment]
8bf07d62dfb2700d2c38822dedac5819e5f4df06 looks great! My comment is a minor thing (but I do want it changed before we merge).

**Comment by uncomputable** [comment]
Rebased and turned `BitWrite::vec` into `crate::write_to_vec`.

**Review by apoelstra — APPROVED** [review]
ACK 5a75b547740f228daa58ae7050fc32302067d68d

**Pull request merged**